src/Test/Integration/DbInitializerAbstract.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace ShoppinPal\YapepCommon\Test\Integration;
+
+
+use YapepBase\Config;
+use YapepBase\Exception\Exception;
+
+/**
+ * Responsible for initializing the Database for integration testing.
+ */
+abstract class DbInitializerAbstract
+{
+
+    /**
+     * @return string
+     */
+    abstract protected function getResourceName();
+
+    /**
+     * @return string
+     */
+    abstract protected function getSchemaFilePath();
+
+
+    public function initDataBase()
+    {
+        $dbConfig = Config::getInstance()->get('resource.database.' . $this->getResourceName() . '.rw.*');
+
+        $testDatabase = $dbConfig['database'];
+
+        $dropDatabaseSql = 'DROP DATABASE IF EXISTS ' . $testDatabase . '; CREATE DATABASE ' . $testDatabase . ';';
+
+        $this->executeSql($dbConfig, '-e ' . escapeshellarg($dropDatabaseSql));
+        $this->executeSql($dbConfig, $testDatabase . ' < ' . escapeshellarg($this->getSchemaFilePath()));
+    }
+
+    /**
+     * @param array  $dbConfig
+     * @param string $commandPart
+     *
+     * @throws \YapepBase\Exception\Exception
+     */
+    protected function executeSql(array $dbConfig, $commandPart)
+    {
+        $charset      = empty($dbConfig['charset']) ? 'utf8' : $dbConfig['charset'];
+        $testHostname = $dbConfig['host'];
+        $testUser     = $dbConfig['user'];
+        $testPassword = $dbConfig['password'];
+
+        $fullSqlCommand = 'mysql'
+            . ' -h ' . $testHostname
+            . ' -u ' . $testUser
+            . ' --default-character-set ' . $charset
+            . ' -p' . escapeshellarg($testPassword)
+            . ' ' . $commandPart
+            . ' 2>&1';
+
+        $output = array();
+        $exitCode = 0;
+        exec($fullSqlCommand, $output, $exitCode);
+
+        foreach ($output as $index => $line) {
+            if ($line == 'mysql: [Warning] Using a password on the command line interface can be insecure.') {
```

Comment:
I know. And sure Mysql can change the string any time. Though they didn't in the past few years, and worst thing what can happen is that change it (I really doubt it) and we have to change it again here or find another way to do this.
I wouldn't waste time on this unless it's really that necessary.